sonar-application/src/main/java/org/sonar/application/AppLogging.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.application;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.ConsoleAppender;
+import ch.qos.logback.core.FileAppender;
+import org.slf4j.LoggerFactory;
+import org.sonar.process.LogbackHelper;
+import org.sonar.process.Props;
+
+/**
+ * Configure logback for the master process
+ */
+class AppLogging {
+
+  static final String CONSOLE_LOGGER = "console";
+  static final String CONSOLE_APPENDER = "CONSOLE";
+  static final String GOBBLER_LOGGER = "gobbler";
+  static final String GOBBLER_APPENDER = "GOBBLER";
+  static final String APP_PATTERN = "%d{yyyy.MM.dd HH:mm:ss} %-5level app[%logger{20}] %msg%n";
+
+  private final LogbackHelper helper = new LogbackHelper();
+
+  LoggerContext configure(Props props) {
+    LoggerContext ctx = helper.getRootContext();
+    ctx.reset();
+    helper.enableJulChangePropagation(ctx);
+    configureConsole(ctx);
+    configureGobbler(props, ctx);
+    configureRoot(ctx);
+    if (props.valueAsBoolean("sonar.log.console", false)) {
+      // used by SonarSource testing environment
+      copyGobblerToConsole();
+    }
+    return ctx;
+  }
+
+  /**
+   * Enable the copy in console of the logs written in logs/sonar.log
+   */
+  private void copyGobblerToConsole() {
```

Comment:
the bootstrap process of sonar-application forks ES and Web processes and forwards all their outputs to file. Gobbler is the name for process output.